src/useStorage/useStorage.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable no-redeclare */
+import { ref, Ref, watchEffect } from 'vue'
+import { StorageManager, StorageType } from './storage'
+
+type UseStorage<T> = {
+  value: Ref<T>,
+  initialValue: T,
+  remove: () => void,
+  set: (value: NonNullable<T>) => void,
+}
+
+export function useStorage<T>(type: StorageType, key: string): UseStorage<T | null>
+export function useStorage<T>(type: StorageType, key: string, defaultValue: T): UseStorage<T>
+export function useStorage<T>(type: StorageType, key: string, defaultValue: T | null = null): UseStorage<T | null> {
+  const storage = new StorageManager(type)
+  const initialValue = storage.get(key, defaultValue)
+  const data = ref(initialValue) as Ref<T | null>
```

Comment:
I think the type that vue is suggesting here is actually more accurate. If I use `useStorage` and my `T` is `Ref<string>`, when you set `data = ref<T>` it will unref `T` to `string`. 

That means your `UseStorage` type should be
```ts
type UseStorage<T> = {
  value: Ref<UnwrapRef<T>>,
  initialValue: T,
  remove: () => void,
  set: (value: NonNullable<UnwrapRef<T>>) => void,
}
```

without this change, if `T` is a ref, ts will mistakingly expect you to be nesting your refs
```ts
const initVal = ref('poop')
const test = useSessionStorage('typescript', initVal)

test.set('sweet') // Argument of type 'string' is not assignable to parameter of type 'Ref<string>'.ts(2345)
```

with the Unref<T> change
```ts
const initVal = ref('poop')
const test = useSessionStorage('typescript', initVal)

test.set('sweet') // works
```

Reply:
Hmm I see the issue, but changing that type I think will cause other issues? Let's look at this sync

Reply:
Update 👍